src/lightkurve/lightcurve.py

```diff
@@ -3140,7 +3206,125 @@ def plot_river(self, **kwargs):
         )
         return ax
+    
 
+    def bin(
+        self,
+        time_bin_size=None,
+        time_bin_start=None,
+        aggregate_func=None,
+        bins=None,
+        n_bins=None,
+
+    ):
+        """Bins a FoldedLightCurve in equally-spaced bins in phase.
+        Binning always occurs in time units (not normalized phase units)
+
+        If the original light curve contains flux uncertainties (``flux_err``),
+        the binned lightcurve will report the root-mean-square error.
+        If no uncertainties are included, the binned curve will return the
+        standard deviation of the data.
+
+        Parameters
+        ----------
+        time_bin_size : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`, optional
+            The time interval for the binned time series - this is either a scalar
+            value (in which case all time bins will be assumed to have the same
+            duration) or as an array of values (in which case each phase bin can
+            have a different duration). If the phase is normalized, the phase will first
+            be converted back into TimeDelta units. If this argument is provided,
+            ``time_bin_end`` should not be provided. 
+            (Default: 0.5 days; default unit: days.)
+        time_bin_start : Time like value, optional
+            The start phase for the binned time series. This can also be a scalar
+            value if ``time_bin_size`` is provided. Defaults to the first
+            time in the sampled time series.
+        aggregate_func : callable, optional
+            The function to use for combining points in the same bin. Defaults
+            to np.nanmean.
+        bins : int, optional
+            int which gives the number of bins to divide the lightkurve into.
+            This adjusts the length of ``time_bin_size``
+            to accommodate the input time series length.
+
+        Returns
+        -------
+        binned_lc : `FoldedLightCurve`
+            A new folded light curve which has been binned, with the 'time' column phase in days 
+        """
+        # astropy's aggregate_downsample function only works when 'time' is type Time or TimeDelta
+        # Since the normalized phase is a unitless quantity, this breaks.
+        # To work around this, we reset the index to be the regular phase (TimeDelta) when binning
+        if n_bins != None:
+            raise ValueError("n_bins is no longer accepted for FoldedLightCurve objects. Please specify 'bins' instead")
+        if bins != None:
+            if not isinstance(bins, int):
+                ValueError('bins must be an integer describing the total number of bins.')
+
+
+        if self.normalize_phase == True:
+            self._replace_normalized_phase()
+
+        if time_bin_start is None:
+            time_bin_start = self.time[0]
+        if np.isscalar(time_bin_size):
+            if isinstance(self.time, TimeDelta):
```

Comment:
The `TimeDelta` check is not necessary, as `self.time` has been temporarily converted to `TimeDelta` (if needed) by the earlier line `self._replace_normalized_phase()`

Reply:
Good catch